src/mono/mono/mini/debugger-agent.c

```diff
@@ -3607,37 +3607,37 @@ init_jit_info_dbg_attrs (MonoJitInfo *ji)
 	if (ji->dbg_attrs_inited)
 		return;
 
+	// NOTE: The following Debugger attributes may not exist if they are trimmed away by the ILLinker
 	MONO_STATIC_POINTER_INIT (MonoClass, hidden_klass)
-		hidden_klass = mono_class_load_from_name (mono_defaults.corlib, "System.Diagnostics", "DebuggerHiddenAttribute");
+		hidden_klass = mono_class_from_name (mono_defaults.corlib, "System.Diagnostics", "DebuggerHiddenAttribute");
 	MONO_STATIC_POINTER_INIT_END (MonoClass, hidden_klass)
```

Comment:
These macros expect the body to return non-null, otherwise they will be executed every time this code is called.

Reply:
Are there other "single time" macros available that set a "has value" flag? - Similar to `Lazy<T>` in C#.

Reply:
I found one: `GENERATE_TRY_GET_CLASS_WITH_CACHE`. Thanks for the tip.